### PR TITLE
Add example to fix sensor clocks

### PR DESCRIPTION
**Pull request by qwandor**
Also changes `BluetoothSession::connect` to wait until services have been discovered before returning.

### CODE REVIEW

File: mijia/examples/fix-clock.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+use tokio_compat_02::FutureExt;
+
+const SCAN_DURATION: Duration = Duration::from_secs(5);
+/// Only correct clocks which are wrong by more than this amount.
+const MINIMUM_OFFSET: Duration = Duration::from_secs(10);
+const CONNECT_TIMEOUT: Duration = Duration::from_secs(10);
+
+/// A duration which may be negative.
+#[derive(Clone, Eq, PartialEq)]
+struct SignedDuration {
```
Comment (alsuren) [inline mijia/examples/fix-clock.rs:23]: feels like you want something like https://docs.rs/utctimestamp/0.1.4/utctimestamp/struct.TimeDelta.html

File: mijia/examples/fix-clock.rs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        }
+        let last_record = session.get_last_history_record(&sensor.id).await?;
+        let sensor_time = session.get_time(&sensor.id).await?;
+        let now = SystemTime::now();
+        let last_record_utc: DateTime<Utc> = last_record.time.into();
+        let sensor_time_utc: DateTime<Utc> = sensor_time.into();
+        let now_utc: DateTime<Utc> = now.into();
+        let offset: SignedDuration = now.duration_since(sensor_time).into();
+        println!(
+            "Time: {} (should be {}, offset {:?}), Last stored value: {}",
```
Comment (alsuren) [inline mijia/examples/fix-clock.rs:100]: (optional) would be interesting to say "last stored value: {}m {}s ago" (which you can calculate using the remote clock)

File: mijia/examples/fix-clock.rs (hunk 1 above)
Comment (alsuren) [reply mijia/examples/fix-clock.rs:23]: turns out we're already depending on chrono, so this wouldn't explode our dependency tree too much.

**Review by alsuren — APPROVED** [review]
Deeply skeptical about the SignedDuration thing. I guess it's okay for an example though.

File: mijia/examples/fix-clock.rs (hunk 2 above)
Comment (qwandor) [reply mijia/examples/fix-clock.rs]: Done.

File: mijia/examples/fix-clock.rs (hunk 1 above)
Comment (qwandor) [reply mijia/examples/fix-clock.rs:23]: It looks about right in intent, but doesn't have the conversion from `Result<Duration, SystemTimeError>` that is the main point.

**Pull request merged**